rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,149 @@
+case-by-case bases for loading components and the parameters that get passed
+into the components.
+
+In most cases, the information coming from this service would help the
+application determine which UI components should **not** be loaded, as applications will
+use the service response for skipping components that aren't needed for
+presenting the data in a screenshot, and thus work towards better performance
+for the display-only purpose.
+
+More background on how Reporting currently works, including the lifecycle of
+creating a PNG report, is here: https://github.com/elastic/kibana/issues/59396
+
+# Basic example
+
+When Kibana loads initially, there is a Newsfeed component in the UI that
+checks internally cached records to see if it must fetch the Elastic News
+Service for newer items. When the Screenshot Mode Service is implemented, the
+Newsfeed component has a source of information to check on whether or not it
+should load in the Kibana UI. If it can avoid loading, it avoids an unnecessary
+HTTP round trip, which weigh heavily on performance.
+
+# Motivation
+
+Kibana Reporting is a commercial feature in Elastic and is highly capable of
+loading the application pages and converting them into a screenshot for PNG or
+PDF export. However, the way it works has downsides with performance,
+maintainability, and expanding it as a tool that can power higher-level
+features. The solution to those downsides is to have Kibana pages themselves
+become more capable at presenting themselves for screenshot capture reports.
+With the Screenshot Mode Service available, Reporting could drop the
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API available as dependency for
+Kibana applications.
+
+A method of the API tells the Application whether or not it should render
+itself to optimize for non-interactivity.
+
+In a future phase, the API might also tell the application more about the
+report, such as PDF page dimensions, or special layout types (print layout,
+etc).
+
+## Interface
+The `setupDeps.screenshotMode` object has a single purpose: tell the app if it
+should render in an optimized way for screenshot capture:
+
+```
+interface IScreenshotModeServiceSetup {
+  isScreenshotMode: () => boolean;
+}
+```
+
+Internally, this object is constructed from a class that refers to information
+sent via a custom proprietary header:
```

Comment:
should we add a possibility of enabling this thru url parameter ? (for easier testing)